server/db/models/player.js

```diff
@@ -148,204 +89,183 @@ playerSchema.methods.setCardinal = function(row, col) {
-  this.set('_id', null); //how should we kill players?
-  this.save();
+playerSchema.methods.isPlayerReady = function() {
+    return this.ready;
 };
 
-playerSchema.methods.gameover = function() {
-  return mongoose.model('Game').findById(this.game)
-  .then(function(game) {
-    game.state = 'gameover';
-    game.save();
-  });
-};
 
-playerSchema.methods.findMyTile = function(row, col){
-  var player = this;
-  var row = row || player.position[0];
-  var col = col || player.postion[1];
+playerSchema.methods.evaluateDamage = function(hitCount) {
+  this.applyDamage(hitCount)
+  this.checkDamage()
+}
 
-  return mongoose.model('Game').findById(player.game)
-  .then(function(game){
-    return mongoose.model('Board').findById(game.board);
-  })
-  .then(function(board){
-    return board.getTileAt(row, col);
-  })
-  .then(function(tileId){
-    return mongoose.model('Tile').findById(tileId);
-  });
-};
 
-playerSchema.methods.attachMyTile = function (){
-  return Game.findById(this.game).bind(this)
-  .then(function(g){
-    return Board.findById(g.board);
-  })
-  .then(function(b){
-    return b.getTileAt(this.position);
-  })
-  .then(function(t){
-    // return player with fully populated tile attached
-    this.tile = t;
-    return this;
-  });
-};
+playerSchema.methods.applyDamage = function(hitCount) {
+  this.damage += hitCount
+}
 
-playerSchema.methods.clearHand = function() {
-    var handToDiscard = this.hand;
-    return mongoose.model('Game').findByIdAndUpdate(player.game,
-        {$push:  {discard: {$each: handToDiscard} } } );
-};
 
-playerSchema.methods.isPlayerReady = function() {
-    return this.ready;
-};
+playerSchema.methods.checkDamage = function() {
+  if (this.damage > 9) {
+    this.loseLife();
+    this.damage = 0
+  }
+}
+
 
 playerSchema.methods.setRegister = function(cards) { //assumes we are getting an array of cards from the front end in order
-  var self = this;
   var prevRegister = this.register;
 
-  if(self.damage < 5 && cards.length === 5) self.register = cards;
-  else if(self.damage === 5 && cards.length === 4) self.register = cards.concat(prevRegister.slice(4));
-  else if(self.damage === 6 && cards.length === 3) self.register = cards.concat(prevRegister.slice(3));
-  else if(self.damage === 7 && cards.length === 2) self.register = cards.concat(prevRegister.slice(2));
-  else if(self.damage === 8 && cards.length === 1) self.register = cards.concat(prevRegister.slice(1));
+  if(this.damage < 5 && cards.length === 5) this.register = cards;
+  else if(this.damage === 5 && cards.length === 4) this.register = cards.concat(prevRegister.slice(4));
+  else if(this.damage === 6 && cards.length === 3) this.register = cards.concat(prevRegister.slice(3));
+  else if(this.damage === 7 && cards.length === 2) this.register = cards.concat(prevRegister.slice(2));
+  else if(this.damage === 8 && cards.length === 1) this.register = cards.concat(prevRegister.slice(1));
   else return;
 
```

Comment:
What if it were something like this:
for (var i === 0 ; i< 5; i++){
   if(prevRegister[i]===0) prevRegister[i]= cards.shift();
}